src/test/java/org/opensearch/knn/index/codec/KNNFormatFactoryTests.java

```diff
@@ -36,4 +36,16 @@ public void testKNN92Format() {
         assertNotNull(knnFormatFacade.compoundFormat());
         assertNotNull(knnFormatFacade.docValuesFormat());
     }
+
+    public void testKNN94Format() {
+        MapperService mapperService = mock(MapperService.class);
+        final Codec lucene94CodecDelegate = KNNCodecFactory.CodecDelegateFactory.createKNN94DefaultDelegate();
+        KNNCodecFactory knnCodecFactory = new KNNCodecFactory(mapperService);
+        final Codec knnCodec = knnCodecFactory.createKNNCodec(lucene94CodecDelegate);
```

Comment:
Why only Codec is final?

Reply:
updated

Reply:
nit: we can make mapperService and knnCodecFactory final as well, but I guess that's ok, we can do it later

Reply:
knnCodecFactory is a constructor call and mapperService is a mock object. Do we need to make it final ?

Reply:
I think it's a preference, make everything final unless you need to mutate that variable, so you'll get the compilation time error. Those two variable can be final, i will not cause issues.

Reply:
My main concern here is inconsistency. I cannot find a reason of why mappserService is not final but Codec is final. I would rather remove final from Codec as well unless we have strong reasoning about it. If you see other test cases, we don't define Codec as final.

Reply:
Agree. To make it consistent, I removed final everywhere. There is no strong reason to have it.